ci-operator/templates/openshift/openshift-ansible/cluster-launch-e2e-40.yaml

```diff
@@ -255,34 +251,80 @@ objects:
 
-        export _CI_ONLY_STAY_AWAY_OPENSHIFT_INSTALL_AWS_USER_TAGS="{\"expirationDate\": \"$(date -d '4 hours' --iso=minutes --utc)\"}"
-        /bin/openshift-install --dir=/tmp/artifacts/installer --log-level=debug create install-config
+        export CLUSTER_ID=$(uuidgen --random)
 
-        # Update install configs to set desired number of masters and workers
-        sed -i "/master/{n;s/1/${MASTERS}/}" /tmp/artifacts/installer/.openshift_install_state.json
-        sed -i "/worker/{n;s/1/${WORKERS}/}" /tmp/artifacts/installer/.openshift_install_state.json
-        sed -i "/master/{n;n;s/1/${MASTERS}/}" /tmp/artifacts/installer/install-config.yml
-        sed -i "/worker/{n;n;s/1/${WORKERS}/}" /tmp/artifacts/installer/install-config.yml
+        if [[ "${CLUSTER_TYPE}" == "gcp" ]]; then
+          cat > /tmp/artifacts/installer/install-config.yml << EOF
+        baseDomain: origin-ci-int-gce.dev.rhcloud.com
+        clusterID:  ${CLUSTER_ID}
+        machines:
+        - name: master
+          replicas: ${MASTERS}
+        - name: worker
+          replicas: ${WORKERS}
+        metadata:
+          name: ${CLUSTER_NAME}
+        networking:
+          clusterNetworks:
+          - cidr:             10.128.0.0/14
+            hostSubnetLength: 9
+          serviceCIDR: 172.30.0.0/16
+          type:        OpenshiftSDN
+        platform:
+          libvirt:
+            URI: qemu+tcp://192.168.122.1/system
+            defaultMachinePlatform:
+              image: file:///unused
+            masterIPs: null
+            network:
+              if: tt0
+              ipRange: 192.168.126.0/24
+        pullSecret: |
+          ${PULL_SECRET}
+        sshKey: |
+          ${SSH_PUB_KEY}
+        EOF
+        elif [[ "${CLUSTER_TYPE}" == "aws" ]]; then
+          export AWS_SHARED_CREDENTIALS_FILE="/etc/openshift-installer/.awscred"
+          export EXPIRATION_DATE=$(date -d '4 hours' --iso=minutes --utc)
+          cat > /tmp/artifacts/installer/install-config.yml << EOF
+        baseDomain: test.ose
+        clusterID:  ${CLUSTER_ID}
+        machines:
+        - name:     master
+          replicas: ${MASTERS}
+        - name:     worker
+          replicas: ${WORKERS}
+        metadata:
+          name: ${CLUSTER_NAME}
+        networking:
+          clusterNetworks:
+          - cidr:             10.128.0.0/14
+            hostSubnetLength: 9
+          serviceCIDR: 172.30.0.0/16
+          type:        OpenshiftSDN
+        platform:
+          aws:
+            region:       us-east-1
+            vpcCIDRBlock: 10.0.0.0/16
+            userTags:
+              expirationDate: ${EXPIRATION_DATE}
+        pullSecret: |
+          ${PULL_SECRET}
+        sshKey: |
+          ${SSH_PUB_KEY}
+        EOF
+        else
+            echo "Unsupported cluster type '${CLUSTER_NAME}'"
+            exit 1
+        fi
 
         /bin/openshift-install --dir=/tmp/artifacts/installer --log-level=debug create ignition-configs &
```

Comment:
`ignition-configs` would destroy `install-config.yml` - is this an installer bug? In any case in CI we should be backing install config since its now cannot be reconstructed using env vars

Reply:
That is core to the design of the installer (so no, not a bug). Why is the install config needed after the Ignition Configs are generated?

Reply:
I'd keep the backup to ensure that the all the vars got templated correctly, this is not required though

Reply:
The installer's state file (`.openshift_install_state.json`) should be archived, so the install-config can be pulled out of there if we need to inspect it after the fact.

Reply:
CI stores this one, but it doesn't seem to contain any config - https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/861/pull-ci-openshift-installer-master-e2e-aws/2219/artifacts/e2e-aws/installer/.openshift_install_state.json - a bug in PR 861?